src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -90,6 +91,27 @@ public function __construct(array $functions)
      * @throws SyntaxError
      */
     public function parse(TokenStream $stream, array $names = [])
+    {
+        $this->lint = false;
+
+        return $this->doParse($stream, $names);
+    }
+
+    /**
+     * Static analysis.
+     *
+     * The syntax of the passed expression will be checked, but not parsed.
+     * If you want to skip checking dynamic variable names, pass `null` instead of the array.
+     *
+     * @throws SyntaxError When passed expression is invalid
+     */
+    public function lint(TokenStream $stream, ?array $names = [])
```

Comment:
let's add `void` as return type then

Reply:
Do I need to immediately add return types for all methods in the Parser class/ExpressionLanguage component? Or is it better to do it in separately PR after merge?

Reply:
Adding return types to existing methods would be a BC break. Only new methods should have them.